src/App.tsx

```diff
@@ -15,71 +19,93 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+export class App extends React.Component<{}, AppState> {
+  state: Readonly<AppState> = {
+    selectedGood: 'Jam',
+  };
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  clearGood = () => {
+    this.setState({ selectedGood: '' });
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  render() {
+    const { selectedGood } = this.state;
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+    const title = selectedGood
+      ? (
+        <h1 className="title is-flex is-align-items-center">
+          {selectedGood}
+          {' '}
+          is selected
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+          <button
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+            onClick={this.clearGood}
+          />
+        </h1>
+      )
+      : (
+        <h1 className="title">No goods selected</h1>
+      );
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+    return (
+      <main className="section container">
+        {title}
+        <table className="table">
+          <tbody>
+            {goods.map((good) => {
+              const isSelected = selectedGood === good;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+              const button = isSelected
```

Comment:
Same here, you could have a separate component that would handle displaying either + or - button, it would receive `isSelected={selectedGood === good}` and  `handleClick={// either clearGood or () => setState}` and then you could just map the values without additional logic like declaring variables

Reply:
![image](https://user-images.githubusercontent.com/78635631/234941086-bdae4229-8fbc-43c2-82e3-0e8715371cb1.png)

Reply:
Aight didn't see that part about not creating additional components.